src/test_l3_rmfAudioCapture.c

```diff
@@ -0,0 +1,1256 @@
+    }
+
+    if (pthread_join(aux_thread, &aux_ret_value) != 0) 
+    {
+        UT_LOG_INFO("Error joining auxiliary capture monitor thread");
+    }
+    if (aux_ret_value != NULL) 
+    {
+        result = *(rmf_Error *)aux_ret_value;
+        UT_ASSERT_EQUAL(result, RMF_SUCCESS);
+    
+        free(aux_ret_value);
+    } else 
+    {
+        UT_LOG_INFO("Thread ret_value NULL, unable to assert for jitter check. Refer prints to confirm if test passed");
+    }
+
+    /* Validate if acceptable level of bytes received */
+    result = validateBytesReceived(&prim_settings, (void *)&prim_ctx, MEASUREMENT_WINDOW_2MINUTES);
+    UT_ASSERT_EQUAL(result, RMF_SUCCESS);
+    result = validateBytesReceived(&aux_settings, (void *)&aux_ctx, MEASUREMENT_WINDOW_2MINUTES);
+    UT_ASSERT_EQUAL(result, RMF_SUCCESS);
+    
+    UT_LOG_INFO("Calling RMF_AudioCapture_Stop(IN:prim_handle:[0x%0X])", &prim_handle);
+    result = RMF_AudioCapture_Stop(prim_handle);
+    UT_LOG_INFO("Result RMF_AudioCapture_Stop(IN:prim_handle:[0x%0X] OUT:rmf_error:[%s]", &prim_handle, UT_Control_GetMapString(rmfError_mapTable, result));
+    prim_ctx.cookie = 0;
+    UT_ASSERT_EQUAL(result, RMF_SUCCESS);
+    UT_LOG_INFO("Calling RMF_AudioCapture_Stop(IN:aux_handle:[0x%0X])", &aux_handle);
+    result = RMF_AudioCapture_Stop(aux_handle);
+    UT_LOG_INFO("Result RMF_AudioCapture_Stop(IN:aux_handle:[0x%0X] OUT:rmf_error:[%s]", &aux_handle, UT_Control_GetMapString(rmfError_mapTable, result));
+    aux_ctx.cookie = 0;
+    UT_ASSERT_EQUAL(result, RMF_SUCCESS);
+
+    sleep(1); // Wait for the last callback to be processed
+    UT_ASSERT_EQUAL(prim_ctx.cookie, 0);
+    UT_ASSERT_EQUAL(aux_ctx.cookie, 0);
+
+    UT_LOG_INFO("Calling RMF_AudioCapture_Close(IN:prim_handle:[0x%0X])", &prim_handle);
+    result = RMF_AudioCapture_Close(prim_handle);
+    UT_LOG_INFO("Result RMF_AudioCapture_Close(IN:prim_handle:[0x%0X] OUT:rmf_error:[%s]", &prim_handle, UT_Control_GetMapString(rmfError_mapTable, result));
+    UT_ASSERT_EQUAL(result, RMF_SUCCESS);
+    UT_LOG_INFO("Calling RMF_AudioCapture_Close(IN:aux_handle:[0x%0X])", &aux_handle);
+    result = RMF_AudioCapture_Close(aux_handle);
+    UT_LOG_INFO("Result RMF_AudioCapture_Close(IN:aux_handle:[0x%0X] OUT:rmf_error:[%s]", &aux_handle, UT_Control_GetMapString(rmfError_mapTable, result));
+
+    UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+
+static UT_test_suite_t * pSuite = NULL;
+
+/**
+ * @brief Register the main tests for this module
+ *
+ * @return int - 0 on success, otherwise failure
+ */
+
+int test_rmfAudioCapture_l3_register(void)
+{
+    // Create the test suite
+    pSuite = UT_add_suite("[L3 rmfAudioCapture]", NULL, NULL);
+    if (pSuite == NULL)
+    {
+        return -1;
+    }
+    // List of test function names and strings
+    UT_add_test(pSuite, "test_l3_rmfAudioCapture_primary_data_check", test_l3_rmfAudioCapture_primary_data_check);
```

Comment:
in the menu test to be select individual API's 
like RMF_AudioCapture_Open_Type
need to list the possible potions when the open select 
 RMF_AC_TYPE_PRIMARY 
  RMF_AC_TYPE_AUXILIARY
  
  please refer other module 
  https://github.com/rdkcentral/rdk-halif-test-device_settings/blob/develop/src/test_l3_dsVideoPort.c#L924
  
  https://github.com/rdkcentral/rdk-halif-test-device_settings/blob/develop/src/test_l3_dsAudio.c#L2296